providers/monitor.rb

```diff
@@ -77,10 +76,12 @@ def whyrun_supported?
 
 private
 
-def yaml_dir
+def conf_yaml_path(resource_name)
   if node['datadog']['agent6']
-    ::File.join(node['datadog']['agent6_config_dir'], 'conf.d')
+    ::File.join(node['datadog']['config_dir'], 'conf.d',
+                "#{resource_name}.d", 'conf.yaml')
   else
-    ::File.join(node['datadog']['config_dir'], 'conf.d')
+    ::File.join(node['datadog']['agent6_config_dir'], 'conf.d',
+                "#{resource_name}.yaml")
```

Comment:
I think the `config_dir` attributes were reverse accidentally here. I think this should be:

```ruby
if node['datadog']['agent6']
  ::File.join(
    node['datadog']['agent6_config_dir'],
    'conf.d',
    "#{resource_name}.d",
    'conf.yaml'
  )
else
  ::File.join(
    node['datadog']['config_dir'],
    'conf.d',
    "#{resource_name}.yaml"
  )
end
```

Reply:
I.e., `git revert 68d00c8463f0144cba6a1186e31ab7c652888362`

Reply:
Ah, it looks like in 68d00c8463f0144cba6a1186e31ab7c652888362 I switched both the base agent config path as well as the resource path; I should have only switched the resource path. So we don't want a full revert. Fixed in 167a877.